### PR TITLE
HPCC-30994 Ensure jobId's removed from log manager in worker

### PR DESCRIPTION
**Pull request by jakesmith**
Also set jobId earlier in containerized, meaning that some early logging is tagged with the correct jobId instead of "UNK".

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by ghalliday — COMMENTED** [review]
A couple of questions - possibly recorded in the jira.

File: thorlcr/slave/thslavemain.cpp [hunk 1]
```diff
@@ -427,9 +435,8 @@ int main( int argc, const char *argv[]  )
 
         if (RegisterSelf(masterEp))
         {
-            if (!slaveLogHandler)
-                slaveLogHandler = startSlaveLog();
-
+            if (globals->getPropBool("@MPChannelReconnect"))
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply thorlcr/slave/thslavemain.cpp:438]: shouldn't here - result of clash mismerge. Will remove.

File: thorlcr/slave/thslavemain.cpp [hunk 2]
```diff
@@ -385,12 +396,9 @@ int main( int argc, const char *argv[]  )
             usage();
 
         mySlaveNum = globals->getPropInt("@slavenum", NotFound);
-        /* NB: in cloud/non-local storage mode, slave number is not known until after registration with the master
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply thorlcr/slave/thslavemain.cpp:388]: this part of the comment would have still been true, 2nd line was misleading.
The comment is now replaced with a more explicit check and exception (line 399-400)

**Comment by jakesmith** [comment]
@ghalliday - please see replies/new commit.

**Review by ghalliday — APPROVED** [review]
Please squash

**Review by ghalliday — APPROVED** [review]
@jakesmith please squash

**Pull request merged**